hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/evm/Evm46ValidationSuite.java

```diff
@@ -1516,16 +1523,81 @@ contract, BALANCE_OF, mirrorAddrWith(systemAccounts.get(i)))
         return defaultHapiSpec("verifiesSystemAccountBalanceOf")
-                .given(cryptoCreate("testAccount").balance(BALANCE), uploadInitCode(contract), contractCreate(contract))
+                .given(cryptoCreate("testAccount").balance(balance), uploadInitCode(contract), contractCreate(contract))
+                .when()
+                .then(opsArray);
+    }
+
+    @HapiTest
+    final HapiSpec testSelfDestructForSystemAccounts() {
+        final AtomicLong deployer = new AtomicLong();
+        final var nonExistingAccountsOps = createOpsArray(
+                nonExistingSystemAccounts,
+                SELF_DESTRUCT_CALLABLE,
+                DESTROY_EXPLICIT_BENEFICIARY,
+                INVALID_SOLIDITY_ADDRESS);
+        final var existingAccountsOps =
+                createOpsArray(existingSystemAccounts, SELF_DESTRUCT_CALLABLE, DESTROY_EXPLICIT_BENEFICIARY, SUCCESS);
+        final var opsArray = new HapiSpecOperation[nonExistingAccountsOps.length + existingAccountsOps.length];
+
+        System.arraycopy(nonExistingAccountsOps, 0, opsArray, 0, nonExistingAccountsOps.length);
+        System.arraycopy(existingAccountsOps, 0, opsArray, nonExistingAccountsOps.length, existingAccountsOps.length);
+
+        return defaultHapiSpec("testSelfDestructForSystemAccounts")
+                .given(
+                        cryptoCreate(BENEFICIARY)
+                                .balance(ONE_HUNDRED_HBARS)
+                                .receiverSigRequired(false)
+                                .exposingCreatedIdTo(id -> deployer.set(id.getAccountNum())),
+                        uploadInitCode(SELF_DESTRUCT_CALLABLE),
+                        contractCreate(SELF_DESTRUCT_CALLABLE).balance(ONE_HBAR))
+                .when()
+                .then(nonExistingAccountsOps);
+    }
+
+    @HapiTest
+    final HapiSpec testSelfDestructForSystemAccountsWithSigRequiredTrue() {
+        final AtomicLong deployer = new AtomicLong();
+        final var nonExistingAccountsOps = createOpsArray(
+                nonExistingSystemAccounts,
+                SELF_DESTRUCT_CALLABLE,
+                DESTROY_EXPLICIT_BENEFICIARY,
+                INVALID_SOLIDITY_ADDRESS);
+        final var existingAccountsOps =
+                createOpsArray(existingSystemAccounts, SELF_DESTRUCT_CALLABLE, DESTROY_EXPLICIT_BENEFICIARY, SUCCESS);
+        final var opsArray = new HapiSpecOperation[nonExistingAccountsOps.length + existingAccountsOps.length];
+
+        System.arraycopy(nonExistingAccountsOps, 0, opsArray, 0, nonExistingAccountsOps.length);
+        System.arraycopy(existingAccountsOps, 0, opsArray, nonExistingAccountsOps.length, existingAccountsOps.length);
+
+        return defaultHapiSpec("testSelfDestructForSystemAccountsWithSigRequiredTrue")
+                .given(
+                        cryptoCreate(BENEFICIARY)
+                                .balance(ONE_HUNDRED_HBARS)
+                                .receiverSigRequired(true)
```

Comment:
Some of the ops should probably fail here, need to sync with @stoqnkpL about this, as I am not 100% sure.

Reply:
Turns out all the system accounts that exist (800-999 from what i've seen) have `receiverSigRequired=false` so the transaction would always succeed.